### DISCUSSION TITLE
Changing dimmer changes the value used for color to 'random' value

### DISCUSSION THREAD

**Discussion by garynl** [arendst/Tasmota#14903] [Q&A] [answered]
I have a LED controller, LS505C-TAS using the latest firmware v11.0.0 Olivia
I intend to control it from Node-Red but have simplified how I use it for this discussion.

I want to change the brightness and the colour of the LEDs to simulate a sunset (for an aquarium). However whenever I change the value for 'dimmer' the result returns a strange value for 'color' and vice versa.

At first I sent the two commands one after another, ut I thought maybe I could resolve the problem using the 'backlog' command to send the 'dimmer' and 'color' values at the same time, but that doesn't help.
Example
...
19:13:12.710 CMD: dimmer 40
19:13:12.719 MQT: stat/LEDSTRIP/RESULT = {"POWER":"ON","Dimmer":40,"Color":"665E04","HSBColor":"55,96,40","Channel":[40,37,1]}
19:13:30.685 CMD: color 8F8F8F
19:13:30.694 MQT: stat/LEDSTRIP/RESULT = {"POWER":"ON","Dimmer":56,"Color":"8F8F8F","HSBColor":"0,0,56","Channel":[56,56,56]}
19:14:29.528 CMD: backlog dimmer 40; color 8F8F8F
19:14:29.552 MQT: stat/LEDSTRIP/RESULT = {"POWER":"ON","Dimmer":40,"Color":"666666","HSBColor":"0,0,40","Channel":[40,40,40]}
19:14:29.760 MQT: stat/LEDSTRIP/RESULT = {"POWER":"ON","Dimmer":56,"Color":"8F8F8F","HSBColor":"0,0,56","Channel":[56,56,56]}
...
The  first result shows the dimmer command was successfully performed. The second one shows the color command was successfully performed, but the value for the dimmer has changed. This causes the LED strip to flash briefly in the first colour before changing it to the second colour. Using the 'backlog' command doesnt help.

Have I misunderstood something? (or not read the documentation carefully enough!)
Is there another way of achieving  changing brightness and colour at the same time?

Any suggestions appreciated.
gary

**Comment by sfromis** [comment, marked as answer]
If dimming and color are mixed up, you may have a bad config, not matching your hardware.

**Reply by garynl** [reply]
Thanks, I'll go through the configuration, probably good to understand it anyway. The controller had Tasmota on it when I bought it, but I did upgrade the firmware (without any problems). I can't say whether the issue was there before I upgraded.

**Comment by garynl** [comment]
The configuration shown on the 'Configure Template' web page is the same as the configuration defined on the Tasmota Supported Devices Repository page. I dont have sufficient knowledge to determine whether the configuration is correct, but I assume so.

**Reply by sfromis** [reply]
Which template is that? The device name you gave is not searchable.

If PWM channels, you can verify r/g/b/c/w channels by activating `SetOption68` to get a slider and toggle per color channel.

**Reply by garynl** [reply]
Sorry,
 its an Athom RGB LED Controller (LS5050C-TAS).  and it does use PWM channels so I will try out your suggestion.
I was reading various articles over the weekend and came to the conclusion that I had a rather naive view of how it worked. I had imagined that the RGB values controlled the colour and the dimmer the brightness so, for example, a dark yellow and a light yellow have the same RGB values, the difference being just the brightness. That s not correct. Our eyes are not equally sensitive to red green and blue and you need to adjust the values to compensate for this. I'm partly colour blind too so I rather depend on the numbers to be sure of a colour; I cant always trust my eyes.

I had intended to adjust the lighting  by changing the brightness and the colour at the same time. I now understand why this will never work so I will have to re-think the way I do it.
Thanks for your help.

**Comment by s-hadinger** [comment]
Using the `color` command forces the output of the PWM. If you want to change only the brightness, use `dimmer` as you used. But if you just want to change the color "hue", use `HSBColor1`.

You can also use `color2 8F8F8F` to change only the color and keep the brightness:
```
CMD: Color 665E04
RSL: RESULT = {"POWER":"ON","Dimmer":40,"Color":"665E04","HSBColor":"55,96,40","Channel":[40,37,1]}

CMD: Color2 8F8F8F
RSL: RESULT = {"POWER":"ON","Dimmer":40,"Color":"666666","HSBColor":"0,0,40","Channel":[40,40,40]}
```

**Reply by garynl** [reply]
Thanks for the tip. I will play with the color2 and HSBColor commands later
